src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -87,130 +96,142 @@ public function handle(AddProductToOrderCommand $command)
-            $result = array_diff($newCartRules, $oldCartRules);
-        } else {
-            $result = [];
-        }
+            if (!empty($newCartRules) && !empty($oldCartRules)) {
+                $result = array_diff($newCartRules, $oldCartRules);
+            } else {
+                $result = [];
+            }
 
-        foreach ($result as $cartRule) {
-            // Create OrderCartRule
-            $rule = new CartRule($cartRule['id_cart_rule']);
-            $values = [
-                'tax_incl' => $rule->getContextualValue(true),
-                'tax_excl' => $rule->getContextualValue(false),
-            ];
-            $orderCartRule = new OrderCartRule();
-            $orderCartRule->id_order = $order->id;
-            $orderCartRule->id_cart_rule = $cartRule['id_cart_rule'];
-            $orderCartRule->id_order_invoice = $invoice->id;
-            $orderCartRule->name = $cartRule['name'];
-            $orderCartRule->value = $values['tax_incl'];
-            $orderCartRule->value_tax_excl = $values['tax_excl'];
-            $orderCartRule->add();
-
-            // @todo: use https://github.com/PrestaShop/decimal
-            $order->total_discounts += $orderCartRule->value;
-            $order->total_discounts_tax_incl += $orderCartRule->value;
-            $order->total_discounts_tax_excl += $orderCartRule->value_tax_excl;
-            $order->total_paid -= $orderCartRule->value;
-            $order->total_paid_tax_incl -= $orderCartRule->value;
-            $order->total_paid_tax_excl -= $orderCartRule->value_tax_excl;
+            foreach ($result as $cartRule) {
+                // Create OrderCartRule
+                $rule = new CartRule($cartRule['id_cart_rule']);
+                $values = [
+                    'tax_incl' => $rule->getContextualValue(true),
+                    'tax_excl' => $rule->getContextualValue(false),
+                ];
+                $orderCartRule = new OrderCartRule();
+                $orderCartRule->id_order = $order->id;
+                $orderCartRule->id_cart_rule = $cartRule['id_cart_rule'];
+                $orderCartRule->id_order_invoice = $invoice->id;
+                $orderCartRule->name = $cartRule['name'];
+                $orderCartRule->value = $values['tax_incl'];
+                $orderCartRule->value_tax_excl = $values['tax_excl'];
+                $orderCartRule->add();
+
+                // @todo: use https://github.com/PrestaShop/decimal
+                $order->total_discounts += $orderCartRule->value;
+                $order->total_discounts_tax_incl += $orderCartRule->value;
+                $order->total_discounts_tax_excl += $orderCartRule->value_tax_excl;
+                $order->total_paid -= $orderCartRule->value;
+                $order->total_paid_tax_incl -= $orderCartRule->value;
+                $order->total_paid_tax_excl -= $orderCartRule->value_tax_excl;
+            }
+
+            $order->update();
+        } catch (\Exception $e) {
```

Comment:
Please add a `use Exception;` instead of the `\` notation

Reply:
Done